File: 2018/python/1.py
def part1(changes):
    print(sum(changes))

def part2(changes):
    seen = {0}
    current = 0
    while True:
        for n in changes:
            current += n
            if current in seen:
                print(current)
                return
            seen.add(current)

if __name__ == "__main__":
    changes = [int(line) for line in open("1.in")]
    part1(changes)
    part2(changes)

File: 2018/lua/1.lua
-- file = io.open('1.in')

-- current = 0
-- seen = {[0] = true}

lines = {}
for n in io.lines("1.in", "n") do
  lines[#lines + 1] = n
end

current = 0
seen = {[0] = true}

while true do
  for i, n in ipairs(lines) do
    current = current + n
    if seen[current] then
      print(current)
      return
    end
    seen[current] = true
  end
end

-- print(current)
